dace/transformation/interstate/state_elimination.py

```diff
@@ -154,11 +154,146 @@ def apply(self, sdfg):
                     break
             else:
                 # If removed assignment does not appear in any other edge,
-                # remove symbol
+                # replace and remove symbol
+                if assignments_to_consider[varname] in sdfg.symbols:
+                    sdfg.replace(varname, assignments_to_consider[varname])
```

Comment:
safe replace of all symbols at the same time?

Reply:
The problem is that safe replace mixes strings and symbols, while SDFG.replace/replace_dict methods require only strings. I will do it with a custom lambda but I think you will see that it is becoming a bit confusing.